### PR TITLE
Formats: Fully deprecate `threads`

### PR DESCRIPTION
**Pull request by KrisXV**

### CODE REVIEW

File: server/chat-commands/info.ts [hunk 1]
```diff
@@ -1923,9 +1914,10 @@ export const commands: Chat.ChatCommands = {
 					rn = rn.split(' ').map((x: string) => x[0].toUpperCase() + x.substr(1)).join(' ');
 					descHtml.push(`&bullet; <a href="${url}">${rn}</a>`);
 				}
-			}
-			if (!descHtml.length && format.threads) {
-				descHtml = format.threads;
+			} else {
+				const genID = ['', 'rb', 'gs', 'rs', 'dp', 'bw', 'xy', 'sm', 'ss', 'sv'];
+				descHtml.push(`This format has no resources linked on its <a href="https://www.smogon.com/dex/${genID[format.gen] || 'sv'}/formats/">Smogon Dex page</a>.` +
```
Comment (mia-pi-git) [inline server/chat-commands/info.ts:1919]:
```suggestion
				const genID = ['rb', 'gs', 'rs', 'dp', 'bw', 'xy', 'sm', 'ss', 'sv'];
				descHtml.push(`This format has no resources linked on its <a href="https://www.smogon.com/dex/${genID[format.gen - 1] || 'sv'}/formats/">Smogon Dex page</a>.` +
```

**Comment by urkerab** [comment]
"Deprecate" is effectively what you did in commit 6cdde67, temporarily restoring the functionality for backward compatibility; this simply removes the support for the property (without actually removing the definition in `dex-threads.ts`).

(Many (436 at time of writing) of my formats link to threads on Smogon but aren't in the API, so I'm going to have to back this out locally anyway.)

**Comment by KrisXV** [comment]
> "Deprecate" is effectively what you did in commit [6cdde67](https://github.com/smogon/pokemon-showdown/commit/6cdde67ed88b47ab6b8ae5717517ae7b5d1ef27b), temporarily restoring the functionality for backward compatibility; this simply removes the support for the property (without actually removing the definition in `dex-threads.ts`).
> 
> (Many (436 at time of writing) of my formats link to threads on Smogon but aren't in the API, so I'm going to have to back this out locally anyway.)

I'll reintroduce support for this then,side servers kinda slipped my mind honestly. sorry!

**Comment by urkerab** [comment]
Cool, thanks!

**Pull request merged**